declarative/declarativemprisplayer.cpp

```diff
@@ -198,6 +204,41 @@ using namespace Amber;
     Defaults to false.
 */
 
+/*!
+    \qmlproperty bool MprisPlayer::canShuffle
+    \brief Indicates whether the player can control the shuffle status
+
+    Should be set to true if the shuffle status of the play queue can be
+    controlled. If set to true, remote controllers can set the shuffle
+    property.
+
+    This value must remain constant during the lifetime of the
+    player and the value can't be changed after the first controller
+    has connected.
+
+    In order to change the value, a new \l MprisPlayer should be
+    instantiated.
+
+    Defaults to false.
```

Comment:
I'm a little bit worried that this defaulting to false will break some users of this library. I already see that apkd doesn't advertise these properties any more, but that we can fix of course.

Reply:
I changed these to true and also added a `reset()` method to `MprisPropertyAdaptor` that's called when the service is unregistered (e.g. by calling `setServiceName(QString())`) so that the properties can be hidden.